### PR TITLE
feat(sdk-coin-hbar): hbar verifyTransaction

### PR DESCRIPTION
**Pull request by jonathancn-bitgo**
BG-49597, BG-49598

### CODE REVIEW

File: modules/sdk-coin-hbar/test/unit/hbar.ts [hunk 1]
```diff
@@ -215,6 +217,183 @@ describe('Hedera Hashgraph:', function () {
...
+      const validTransaction = await basecoin.verifyTransaction({
+        txParams,
+        txPrebuild,
+        memo,
+        wallet: walletObj,
+      } as any);
+      validTransaction.should.equal(true);
+    });
+
+    it('should verify create associated token account transaction', async function () {
```
Comment (joey-bai) [inline modules/sdk-coin-hbar/test/unit/hbar.ts:351]: could you add a failure case for this?

File: modules/sdk-coin-hbar/test/unit/hbar.ts [hunk 2]
```diff
@@ -215,6 +217,183 @@ describe('Hedera Hashgraph:', function () {
...
+      const validTransaction = await basecoin.verifyTransaction({
+        txParams,
+        txPrebuild,
+        memo,
+        wallet: walletObj,
+      } as any);
+      validTransaction.should.equal(true);
+    });
+
+    it('should verify token transfer transaction', async function () {
```
Comment (joey-bai) [inline modules/sdk-coin-hbar/test/unit/hbar.ts:371]: same, add failure case

File: modules/sdk-coin-hbar/test/unit/hbar.ts [hunk 3]
```diff
@@ -215,6 +217,183 @@ describe('Hedera Hashgraph:', function () {
...
+      };
+      newTxParamsWithError = () => {
+        return _.cloneDeep(txParamsWithError);
+      };
+      newTxParamsWithExtraData = () => {
+        return _.cloneDeep(txParamsWithExtraData);
+      };
+    });
+
+    it('should verify transactions', async function () {
```
Comment (joey-bai) [inline modules/sdk-coin-hbar/test/unit/hbar.ts:310]: this is for native transfer right?

File: modules/sdk-coin-hbar/test/unit/hbar.ts (hunk 2 above)
Comment (joey-bai) [reply modules/sdk-coin-hbar/test/unit/hbar.ts:371]: maybe something around token name mismatch

**Review by joey-bai — APPROVED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**